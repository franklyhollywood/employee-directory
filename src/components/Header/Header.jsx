export default function Header() {
  return (
    <div>
      acme inc name, sign in button if user is signed out, you are signed in as:
      email, signout button if user signed in
    </div>
  );
}
