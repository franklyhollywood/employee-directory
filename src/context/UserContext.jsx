import { useContext, createContext, useState } from 'react';

const userContext = createContext();

const userProvider = ({ children }) => {
  const [user, setUser] = useState('');

  return (
    <userContext.Provider value={{ user, setUser }}>
      {children}
    </userContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(userContext);

  if (context === undefined) {
    throw new Error('Incorrect Placement');
  }
  return context;
};

export { userProvider, useUser };
