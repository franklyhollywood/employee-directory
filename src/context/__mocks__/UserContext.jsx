import { useContext, createContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(mockUser ? { email: mockUser.email } : {});

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const UserCtx = useContext(UserContext);

  if (UserCtx === undefined) {
    throw new Error('Must be wrapped in a userprovider to use useUser');
  }
  return UserCtx;
};

export { UserProvider, useUser };
