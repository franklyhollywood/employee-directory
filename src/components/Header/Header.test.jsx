import { render } from '@testing-library/react';
import Header from './Header';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext.jsx';

jest.mock('../../services/users.js');
jest.mock('../../context/UserContext.jsx');

it('header test', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});

it('header test', () => {
  const { container } = render(
    <UserProvider mockUser={{ email: 'email@email.com' }}>
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
