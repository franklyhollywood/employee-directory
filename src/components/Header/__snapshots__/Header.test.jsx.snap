// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header test 1`] = `
<div>
  <div>
    <div>
      <div>
         
        <button>
          sign out
        </button>
      </div>
    </div>
    <h1>
      Acme Employee Directory
    </h1>
    <hr />
  </div>
</div>
`;

exports[`header test 2`] = `
<div>
  <div>
    <div>
      <div>
        email@email.com
         
        <button>
          sign out
        </button>
      </div>
    </div>
    <h1>
      Acme Employee Directory
    </h1>
    <hr />
  </div>
</div>
`;
